templates/CRM/Contribute/Form/Contribution.tpl

```diff
@@ -319,6 +319,16 @@
             <td {$valueStyle}>{$form.payment_instrument_id.html} {help id="payment_instrument_id"}</td>
             </td>
           </tr>
+          {if $form.card_type}
+            <tr id="cardType" class="crm-contribution-form-block-card_type">
+              <td class="label">{$form.card_type.label}</td>
+              <td {$valueStyle}>{$form.card_type.html}</td>
+            </tr>
+            <tr id="cardNumber" class="crm-contribution-form-block-credit_card_number">
```

Comment:
@pradpnayak please reply to @eileenmcnaughton indicating that you have accepted her point about being too tricksy a style and will make a change to more straightforward coding approach or why you don't think it is necessary.